File: resources/ios/Sources/MobikulLoaderFunctions.swift
import Foundation
import NativePHP

enum MobikulLoaderFunctions {
    struct Show: BridgeFunction {
        func run(parameters: [String: Any], resolve: ResolveBlock, reject: RejectBlock) {
            let message = (parameters["message"] as? String) ?? "Loading..."
            resolve([
                "visible": true,
                "message": message
            ])
        }
    }

    struct Hide: BridgeFunction {
        func run(parameters: [String: Any], resolve: ResolveBlock, reject: RejectBlock) {
            resolve([
                "visible": false
            ])
        }
    }
}

File: resources/android/src/MobikulLoaderFunctions.kt
package com.mobikul.plugins.loader

import androidx.fragment.app.FragmentActivity
import com.nativephp.mobile.bridge.BridgeFunction
import com.nativephp.mobile.bridge.BridgeResponse

object MobikulLoaderFunctions {
    class Show(private val activity: FragmentActivity) : BridgeFunction {
        override fun execute(parameters: Map<String, Any>): Map<String, Any> {
            return BridgeResponse.success(
                mapOf(
                    "visible" to true,
                    "message" to (parameters["message"] ?: "Loading...")
                )
            )
        }
    }

    class Hide(private val activity: FragmentActivity) : BridgeFunction {
        override fun execute(parameters: Map<String, Any>): Map<String, Any> {
            return BridgeResponse.success(mapOf("visible" to false))
        }
    }
}
